tts/aws/src/lib.rs

```diff
@@ -0,0 +1,1153 @@
+
+        Ok(SynthesisStream::new(PollySynthesisStream::new(
+            voice_id, client, options,
+        )))
+    }
+
+    fn create_voice_conversion_stream(
+        target_voice: golem_tts::golem::tts::voices::VoiceBorrow<'_>,
+        _options: Option<SynthesisOptions>,
+    ) -> Result<VoiceConversionStream, TtsError> {
+        let client = Self::create_client()?;
+        let voice_id = target_voice.get::<PollyVoiceImpl>().get_id();
+
+        Ok(VoiceConversionStream::new(PollyVoiceConversionStream::new(
+            voice_id, client,
+        )))
+    }
+}
+
+impl AdvancedGuest for AwsPollyComponent {
+    type PronunciationLexicon = PollyPronunciationLexicon;
+    type LongFormOperation = PollyLongFormOperation;
+
+    fn create_voice_clone(
+        _name: String,
+        _audio_samples: Vec<AudioSample>,
+        _description: Option<String>,
+    ) -> Result<golem_tts::golem::tts::voices::Voice, TtsError> {
+        Err(TtsError::UnsupportedOperation(
+            "Voice cloning not supported by AWS Polly".to_string(),
+        ))
+    }
+
+    fn design_voice(
+        _name: String,
+        _characteristics: VoiceDesignParams,
+    ) -> Result<golem_tts::golem::tts::voices::Voice, TtsError> {
+        Err(TtsError::UnsupportedOperation(
+            "Voice design not supported by AWS Polly".to_string(),
+        ))
+    }
+
+    fn convert_voice(
+        _input_audio: Vec<u8>,
+        _target_voice: golem_tts::golem::tts::voices::VoiceBorrow<'_>,
+        _preserve_timing: Option<bool>,
+    ) -> Result<Vec<u8>, TtsError> {
+        Err(TtsError::UnsupportedOperation(
+            "Voice conversion not supported by AWS Polly".to_string(),
+        ))
+    }
+
+    fn generate_sound_effect(
+        _description: String,
+        _duration_seconds: Option<f32>,
+        _style_influence: Option<f32>,
+    ) -> Result<Vec<u8>, TtsError> {
+        Err(TtsError::UnsupportedOperation(
+            "Sound effect generation not supported by AWS Polly".to_string(),
+        ))
+    }
+
+    fn create_lexicon(
+        name: String,
+        language: LanguageCode,
+        entries: Option<Vec<PronunciationEntry>>,
+    ) -> Result<PronunciationLexicon, TtsError> {
+        let client = Self::create_client()?;
+        let lexicon = PollyPronunciationLexicon::new(name, language, entries, client);
+
+        lexicon.sync_lexicon()?;
+
+        Ok(PronunciationLexicon::new(lexicon))
+    }
+
+    fn synthesize_long_form(
+        content: String,
+        voice: golem_tts::golem::tts::voices::VoiceBorrow<'_>,
+        output_location: String,
+        _chapter_breaks: Option<Vec<u32>>,
+    ) -> Result<LongFormOperation, TtsError> {
+        let client = Self::create_client()?;
+        let voice_id = voice.get::<PollyVoiceImpl>().get_id();
+        let voice_language = voice.get::<PollyVoiceImpl>().get_language();
+
+        let bucket_name = if output_location.starts_with("s3://") {
```

Comment:
Let's fail with an error if `output_location` is not in the proper format or does not have a bucket name. No default-bucket.